src/Language/PureScript/TypeChecker/Types.hs

```diff
@@ -477,7 +477,9 @@ inferBinder val (ConstructorBinder ss ctor binders) = do
       (_, fn) <- instantiatePolyTypeWithUnknowns (internalError "Data constructor types cannot contain constraints") ty
       fn' <- introduceSkolemScope <=< replaceAllTypeSynonyms $ fn
       let (args, ret) = peelArgs fn'
-      unless (length args == length binders) . throwError . errorMessage $ IncorrectConstructorArity ctor
+          expected = length args
```

Comment:
Are these two the right way around? I think they are, but I'm not 100% sure.

Reply:
Yeah I'm pretty sure this is right, since `expected` comes from the environment (a few steps removed), whereas `binders` is from the case match.